### Heroes Of Pymoli Data Analysis
* Of the 1163 active players, the vast majority are male (84%). There also exists, a smaller, but notable proportion of female players (14%).

* Our peak age demographic falls between 20-24 (44.8%) with secondary groups falling between 15-19 (18.60%) and 25-29 (13.4%).  
-----

### Note
* Instructions have been included for each segment. You do not have to follow them exactly, but they are included to help you think through the steps.

In [223]:
# Dependencies and Setup
import pandas as pd
import numpy as np

# File to Load (Remember to Change These)
file_to_load = "Resources/purchase_data.csv"

# Read Purchasing File and store into Pandas data frame
purchase_data = pd.read_csv(file_to_load)

## Player Count

* Display the total number of players


In [224]:
# Show the first five rows of the dataset

purchase_data.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [225]:
#The number of players
player_count= purchase_data['SN'].nunique()
print(f'There are {player_count} players in this dataset')

There are 576 players in this dataset


In [226]:
# Convert a list of dictionarys into a dataframe
total_players=[{'Total Players':player_count}]
pd.DataFrame(total_players)


Total Players
0            576

## Purchasing Analysis (Total)

* Run basic calculations to obtain number of unique items, average price, etc.


* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame


In [227]:
#Number of unique Items
Item_count=purchase_data['Item ID'].nunique()
print(f'There are {Item_count} unique items.')

There are 183 unique items.


In [228]:
# Average price
ave_price=round(purchase_data['Price'].mean(),2)
print(f'The average price is ${ave_price}')

The average price is $3.05


In [229]:
# Number of Purchases
purchases_count=purchase_data['Purchase ID'].count()
print(f'There are total {purchases_count} purchase in this dataset')

There are total 780 purchase in this dataset


In [230]:
#Total Revenue
total_revenue =purchase_data['Price'].sum()
print(f'The total revenue is ${total_revenue}.')

The total revenue is $2379.77.


In [231]:
# Convert a list of dictionarys into a dataframe
purchasing_ana=[{'Number of Unique Items':Item_count,
                 'Average Price':ave_price,'Number of Purchases':purchases_count,
                 'Total Revenue':total_revenue}]
purchasing_ana_df = pd.DataFrame(purchasing_ana)


In [232]:
#Format the dataframe and reorgnize the columns
purchasing_ana_df['Average Price']=purchasing_ana_df['Average Price'].map("${:.2f}".format)
purchasing_ana_df['Total Revenue']=purchasing_ana_df['Total Revenue'].map("${:.2f}".format)

purchasing_ana_df= purchasing_ana_df[['Number of Unique Items','Average Price',
                                      'Number of Purchases','Total Revenue']]

purchasing_ana_df

Number of Unique Items Average Price  Number of Purchases Total Revenue
0                     183         $3.05                  780      $2379.77

## Gender Demographics

* Percentage and Count of Male Players


* Percentage and Count of Female Players


* Percentage and Count of Other / Non-Disclosed




In [252]:
# Extract 'SN' and 'Gender' two columns from original dataframe and save to a new dateframe
# Drop the duplicated rows(the player listed repeatedly)

Gender_ana = purchase_data[['SN','Gender']].drop_duplicates()
# Count the male players and female players
gender_count = Gender_ana['Gender'].value_counts()
gender_count

Male                     484
Female                    81
Other / Non-Disclosed     11
Name: Gender, dtype: int64

In [253]:
# Count of  gender  percentage
gender_percent = Gender_ana['Gender'].value_counts()*100/player_count


In [255]:
# Creat a dataframe
gender_df = gender_info = pd.DataFrame({"Total Count": gender_count,
                          "Percentage of Players": gender_percent})
# Format value in "Percentage of Players" column
gender_df["Percentage of Players"] = gender_df["Percentage of Players"].map("{:.2f}".format)

gender_df

Total Count Percentage of Players
Male                           484                 84.03
Female                          81                 14.06
Other / Non-Disclosed           11                  1.91


## Purchasing Analysis (Gender)

* Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. by gender




* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame

In [257]:
# group dataframe by Gender and count the total purchase value
total_value= pd.DataFrame(purchase_data.groupby('Gender')['Price'].sum())
# Rename column name
total_value= total_value.rename(columns={'Price':'Total Purchase Value'})
total_value

Total Purchase Value
Gender                                     
Female                               361.94
Male                                1967.64
Other / Non-Disclosed                 50.19

In [258]:
# group dataframe by Gender and count the purchase times
purchase_count = pd.DataFrame(purchase_data.groupby('Gender')['Price'].count())
# Rename column name
purchase_count = purchase_count.rename(columns={'Price':'Purchase Count'})
purchase_count

Purchase Count
Gender                               
Female                            113
Male                              652
Other / Non-Disclosed              15

In [259]:
# group dataframe by Gender and count the average price
ave_price = pd.DataFrame(purchase_data.groupby('Gender')['Price'].mean())
# Rename column name
ave_price = ave_price.rename(columns={'Price':'Average Purchase Price'})
ave_price

Average Purchase Price
Gender                                       
Female                               3.203009
Male                                 3.017853
Other / Non-Disclosed                3.346000

In [266]:
# Combine four  dataframe 'total_value' ,'purchase_count','ave_price' ,'gender_count'
df_combine=total_value.join(purchase_count,how='left')
df_combine = df_combine.join(ave_price, how='left')
df_combine=df_combine.join(pd.DataFrame(gender_count), how = 'left')
df_combine

Total Purchase Value  Purchase Count  \
Gender                                                        
Female                               361.94             113   
Male                                1967.64             652   
Other / Non-Disclosed                 50.19              15   

                       Average Purchase Price  Gender  
Gender                                                 
Female                               3.203009      81  
Male                                 3.017853     484  
Other / Non-Disclosed                3.346000      11

In [267]:
# Caculating purchase total per person 
df_combine["Avg Total Purchase per Person"] = df_combine["Total Purchase Value"]/df_combine["Gender"]

In [268]:
# Format 'Total Purchase Value' column , 'Avg Total Purchase per Person' column and AVerage Purchase Price' column
columns=['Total Purchase Value' , 'Avg Total Purchase per Person','Average Purchase Price']
for c in columns:
    df_combine[c]= df_combine[c].map("${:.2f}".format)
# Drop columns 'Percentage of Players' and 'Total Count' and reorganize the columns
df_combine=df_combine[['Purchase Count', 'Average Purchase Price','Total Purchase Value',
                       'Avg Total Purchase per Person']]
# Check the change 
df_combine


Purchase Count Average Purchase Price  \
Gender                                                         
Female                            113                  $3.20   
Male                              652                  $3.02   
Other / Non-Disclosed              15                  $3.35   

                      Total Purchase Value Avg Total Purchase per Person  
Gender                                                                    
Female                             $361.94                         $4.47  
Male                              $1967.64                         $4.07  
Other / Non-Disclosed               $50.19                         $4.56

## Age Demographics

* Establish bins for ages


* Categorize the existing players using the age bins. Hint: use pd.cut()


* Calculate the numbers and percentages by age group


* Create a summary data frame to hold the results


* Optional: round the percentage column to two decimal points


* Display Age Demographics Table


In [318]:
# Check the 'Age' data distribution 
purchase_data['Age'].describe()

count    780.000000
mean      22.714103
std        6.659444
min        7.000000
25%       20.000000
50%       22.000000
75%       25.000000
max       45.000000
Name: Age, dtype: float64

In [319]:
# Create bins in which to place values based on 'Age'
bins=[0,9,14,19,24,29,34,39,50]
# Create labels for these bins
group_labels = ["<10", "10-14", "15-19", "20-24", "25-29",
                "30-34","35-39", "40+"]

In [320]:
# Add a new column"Age Group" to lable the age
purchase_data["Age Group"] = pd.cut(purchase_data["Age"], bins, labels=group_labels)
purchase_data.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price Age Group  
0  Extraction, Quickblade Of Trembling Hands   3.53     20-24  
1                          Frenzied Scimitar   1.56       40+  
2                               Final Critic   4.88     20-24  
3                                Blindscythe   3.27     20-24  
4                                       Fury   1.44     20-24

In [321]:
# Retrieve the SN, Age, and 'Age Group columns
age_ana_df = purchase_data[['SN','Age','Age Group']]
# Remove the duplicated rows
age_ana_df = age_ana_df.drop_duplicates()


In [322]:
# Calculate the numbers by age group
age_df = pd.DataFrame(age_ana_df["Age Group"].value_counts())

age_df

Age Group
20-24        258
15-19        107
25-29         77
30-34         52
35-39         31
10-14         22
<10           17
40+           12

In [323]:
# Calculate the percentage by age group

age_df['Percentage of Players']=round(age_df['Age Group']/player_count*100,2)

In [324]:
# Rename column 
age_df=age_df.rename(columns={'Age Group':'Total Count'})
# Sort rows by index
age_df.sort_index(axis=0,  ascending=True)

Total Count  Percentage of Players
<10             17                   2.95
10-14           22                   3.82
15-19          107                  18.58
20-24          258                  44.79
25-29           77                  13.37
30-34           52                   9.03
35-39           31                   5.38
40+             12                   2.08

## Purchasing Analysis (Age)

* Bin the purchase_data data frame by age


* Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. in the table below


* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame

In [325]:
# Calculate the purchase value by age group, save the values as a list
Total_value_per_group = list(purchase_data.groupby('Age Group').Price.sum().values)
Total_value_per_group

[77.13,
 82.77999999999999,
 412.8899999999998,
 1114.0600000000006,
 292.99999999999983,
 214.0,
 147.67,
 38.24]

In [326]:
# Calculate the average purchase value by age group and  save the values as a list
average_purch_list =list(purchase_data.groupby('Age Group').Price.mean().values)
average_purch_list 

[3.353478260869565,
 2.956428571428571,
 3.03595588235294,
 3.0522191780821935,
 2.900990099009899,
 2.9315068493150687,
 3.6017073170731706,
 2.9415384615384617]

In [327]:
# Calculate the purchase times by age group and save the values as a list
purch_count = list(purchase_data.groupby('Age Group').Price.count().values)
purch_count

[23, 28, 136, 365, 101, 73, 41, 13]

In [328]:
# Calculate the Average total purchase per person and save the values as a list
# Average total purchase per person = total purchase value in each age group/ total number of player in each age group
avg_purch_per_person_list = purchase_data.groupby('Age Group').Price.sum().values/age_df['Total Count'].sort_index(axis=0,  ascending=True).values
avg_purch_per_person_list =list(avg_purch_per_person_list)
avg_purch_per_person_list

[4.537058823529412,
 3.7627272727272723,
 3.85878504672897,
 4.3180620155038785,
 3.805194805194803,
 4.115384615384615,
 4.763548387096773,
 3.186666666666667]

In [329]:
# Convert all data to a dataframe
raw_data_info = {'Purchase Count':purch_count,'Average Purchase Price':average_purch_list,
                  'Total Purchase Value':Total_value_per_group,
                 'Avg Total Purchase per Person':avg_purch_per_person_list}

df_columns=['Purchase Count','Average Purchase Price','Total Purchase Value','Avg Total Purchase per Person']
df_index=list(purchase_data.groupby('Age Group').Price.sum().keys())
purchase_age_df = pd.DataFrame(raw_data_info,columns=df_columns,index=df_index)

In [330]:
# Show the dataframe
purchase_age_df

Purchase Count  Average Purchase Price  Total Purchase Value  \
<10                23                3.353478                 77.13   
10-14              28                2.956429                 82.78   
15-19             136                3.035956                412.89   
20-24             365                3.052219               1114.06   
25-29             101                2.900990                293.00   
30-34              73                2.931507                214.00   
35-39              41                3.601707                147.67   
40+                13                2.941538                 38.24   

       Avg Total Purchase per Person  
<10                         4.537059  
10-14                       3.762727  
15-19                       3.858785  
20-24                       4.318062  
25-29                       3.805195  
30-34                       4.115385  
35-39                       4.763548  
40+                         3.186667

In [331]:
# Organize the rows, put the first row(index="<10") to the last one
purchase_age_df=purchase_age_df[1:].append(purchase_age_df[:1])


In [333]:
# Format the value in 'Average Purchase Price','Total Purchase Value',
# 'Avg Total Purchase per Person' columns
columns=['Average Purchase Price','Total Purchase Value','Avg Total Purchase per Person']
for c in columns:
    purchase_age_df[c] = purchase_age_df[c].map("${:.2f}".format)    
   

ValueError: Unknown format code 'f' for object of type 'str'

In [334]:
purchase_age_df

Purchase Count Average Purchase Price Total Purchase Value  \
10-14              28                  $2.96               $82.78   
15-19             136                  $3.04              $412.89   
20-24             365                  $3.05             $1114.06   
25-29             101                  $2.90              $293.00   
30-34              73                  $2.93              $214.00   
35-39              41                  $3.60              $147.67   
40+                13                  $2.94               $38.24   
<10                23                  $3.35               $77.13   

      Avg Total Purchase per Person  
10-14                         $3.76  
15-19                         $3.86  
20-24                         $4.32  
25-29                         $3.81  
30-34                         $4.12  
35-39                         $4.76  
40+                           $3.19  
<10                           $4.54

## Top Spenders

* Run basic calculations to obtain the results in the table below


* Create a summary data frame to hold the results


* Sort the total purchase value column in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the summary data frame



In [291]:
# Calculate the purchase times by SN group
Purchase_count = pd.DataFrame(purchase_data.groupby('SN').count().Price)
# Rename the column
Purchase_count = Purchase_count.rename(columns={'Price':'Purchase Count'})
Purchase_count.head()

Purchase Count
SN                          
Adairialis76               1
Adastirin33                1
Aeda94                     1
Aela59                     1
Aelaria33                  1

In [292]:
# Calculate the total purchase value by SN group
total_value = pd.DataFrame(purchase_data.groupby('SN').sum().Price)
# Rename the column
total_value = total_value.rename(columns={'Price':'Total Purchase Value'})

total_value.head()

Total Purchase Value
SN                                
Adairialis76                  2.28
Adastirin33                   4.48
Aeda94                        4.91
Aela59                        4.32
Aelaria33                     1.79

In [293]:
# Combine two dataframe Purchase_count and total_value
top_spender_df = Purchase_count.join(total_value,how='left')


In [294]:
# Sort the total purchase value column in descending order
# Slice the dataframe only keep the first five rows
top_spender_df = top_spender_df.sort_values(by=['Total Purchase Value'],ascending=False)[:5]



In [295]:
# Calculate the average purchase price
top_spender_df['Average Purchase Price']= top_spender_df['Total Purchase Value']/top_spender_df['Purchase Count']
top_spender_df

Purchase Count  Total Purchase Value  Average Purchase Price
SN                                                                       
Lisosia93                 5                 18.96                3.792000
Idastidru52               4                 15.45                3.862500
Chamjask73                3                 13.83                4.610000
Iral74                    4                 13.62                3.405000
Iskadarya95               3                 13.10                4.366667

In [296]:
# Merge top_spender_df with the original dataframe to get more infomation about these five spender.
df=pd.merge(top_spender_df,purchase_data,on='SN',how='left')[['SN','Age','Gender']]
# drop the duplicated rows
df=df.drop_duplicates()
df

SN  Age  Gender
0     Lisosia93   25    Male
5   Idastidru52   24    Male
9    Chamjask73   22  Female
12       Iral74   21    Male
16  Iskadarya95   20    Male

In [297]:
# Reorgnize the columns and format the dataframe 
top_spender_df = top_spender_df[['Purchase Count','Average Purchase Price','Total Purchase Value']]
top_spender_df['Total Purchase Value']=top_spender_df['Total Purchase Value'].map('${:.2f}'.format)
top_spender_df['Average Purchase Price']=top_spender_df['Average Purchase Price'].map('${:.2f}'.format)
top_spender_df


Purchase Count Average Purchase Price Total Purchase Value
SN                                                                     
Lisosia93                 5                  $3.79               $18.96
Idastidru52               4                  $3.86               $15.45
Chamjask73                3                  $4.61               $13.83
Iral74                    4                  $3.40               $13.62
Iskadarya95               3                  $4.37               $13.10

## Most Popular Items

* Retrieve the Item ID, Item Name, and Item Price columns


* Group by Item ID and Item Name. Perform calculations to obtain purchase count, item price, and total purchase value


* Create a summary data frame to hold the results


* Sort the purchase count column in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the summary data frame



In [310]:
# Retrieve the Item ID, Item Name, and Item Price columns
Popular_Items_df=purchase_data[['Item ID','Item Name','Price']]
Popular_Items_df.head()

Item ID                                  Item Name  Price
0      108  Extraction, Quickblade Of Trembling Hands   3.53
1      143                          Frenzied Scimitar   1.56
2       92                               Final Critic   4.88
3      100                                Blindscythe   3.27
4      131                                       Fury   1.44

In [311]:
# Calculate the total purchase value by 'Item ID','Item Name' columns
Popular_Items_df= Popular_Items_df.groupby(['Item ID','Item Name'])['Price'].value_counts()
# Convert to dataframe
Popular_Items_df= pd.DataFrame(Popular_Items_df)
# Remane column and convert the Price index to column
Popular_Items_df = Popular_Items_df.rename(columns={'Price':'Purchase Count'})
Popular_Items_df = Popular_Items_df.reset_index(level=['Price'])


Popular_Items_df.head()


Price  Purchase Count
Item ID Item Name                                
0       Splinter             1.28               4
1       Crucifer             3.26               3
2       Verdict              2.48               6
3       Phantomlight         2.49               6
4       Bloodlord's Fetish   1.70               5

In [312]:
# Remane column
Popular_Items_df = Popular_Items_df.rename(columns={'Price':'Item Price'})
# Calculate the total purchase value
Popular_Items_df['Total Purchase Value']= (Popular_Items_df['Item Price'])* (Popular_Items_df['Purchase Count'])


In [313]:
# Format the value
Popular_Items_df['Item Price']=Popular_Items_df['Item Price'].map("${:.2f}".format)    
Popular_Items_df['Total Purchase Value']=Popular_Items_df['Total Purchase Value'].map("${:.2f}".format)

In [314]:
# Reorganize the columns name
Popular_Items_df= Popular_Items_df[['Purchase Count','Item Price','Total Purchase Value']]
# Sort the purchase count column in descending order
Popular_Items_df= Popular_Items_df. sort_values(by='Purchase Count', ascending=False)
Popular_Items_df.head()

Purchase Count  \
Item ID Item Name                                                      
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
108     Extraction, Quickblade Of Trembling Hands                  9   
82      Nirvana                                                    9   
19      Pursuit, Cudgel of Necromancy                              8   

                                                     Item Price  \
Item ID Item Name                                                 
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
145     Fiery Glass Crusader                              $4.58   
108     Extraction, Quickblade Of Trembling Hands         $3.53   
82      Nirvana                                           $4.90   
19      Pursuit, Cudgel of Necromancy                     $1.02   

                                                     Total Purchase Value  
Item ID Item Name                                                          
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
108     Extraction, Quickblade Of Trembling Hands                  $31.77  
82      Nirvana                                                    $44.10  
19      Pursuit, Cudgel of Necromancy                               $8.16

## Most Profitable Items

* Sort the above table by total purchase value in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the data frame



In [214]:
Popular_Items_df.head()

Purchase Count  \
Item ID Item Name                                                      
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
108     Extraction, Quickblade Of Trembling Hands                  9   
82      Nirvana                                                    9   
19      Pursuit, Cudgel of Necromancy                              8   

                                                     Item Price  \
Item ID Item Name                                                 
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
145     Fiery Glass Crusader                              $4.58   
108     Extraction, Quickblade Of Trembling Hands         $3.53   
82      Nirvana                                           $4.90   
19      Pursuit, Cudgel of Necromancy                     $1.02   

                                                     Total Purchase Value  
Item ID Item Name                                                          
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
108     Extraction, Quickblade Of Trembling Hands                  $31.77  
82      Nirvana                                                    $44.10  
19      Pursuit, Cudgel of Necromancy                               $8.16

In [215]:
# Convert the datatype from string to float in 'Total Purchase Value' column

Popular_Items_df['Total Purchase Value'] = Popular_Items_df['Total Purchase Value'].apply(lambda x:pd.to_numeric(x[1:]))


In [216]:
# check the change
Popular_Items_df.head()

Purchase Count  \
Item ID Item Name                                                      
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
108     Extraction, Quickblade Of Trembling Hands                  9   
82      Nirvana                                                    9   
19      Pursuit, Cudgel of Necromancy                              8   

                                                     Item Price  \
Item ID Item Name                                                 
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
145     Fiery Glass Crusader                              $4.58   
108     Extraction, Quickblade Of Trembling Hands         $3.53   
82      Nirvana                                           $4.90   
19      Pursuit, Cudgel of Necromancy                     $1.02   

                                                      Total Purchase Value  
Item ID Item Name                                                           
178     Oathbreaker, Last Hope of the Breaking Storm                 50.76  
145     Fiery Glass Crusader                                         41.22  
108     Extraction, Quickblade Of Trembling Hands                    31.77  
82      Nirvana                                                      44.10  
19      Pursuit, Cudgel of Necromancy                                 8.16

In [217]:
# Sort the dataframe by total purchase value in descending order
Profitable_Items_df = Popular_Items_df.sort_values(by='Total Purchase Value',ascending=False)

In [218]:
# Check the change
Profitable_Items_df.head()

Purchase Count  \
Item ID Item Name                                                      
178     Oathbreaker, Last Hope of the Breaking Storm              12   
82      Nirvana                                                    9   
145     Fiery Glass Crusader                                       9   
92      Final Critic                                               8   
103     Singed Scalpel                                             8   

                                                     Item Price  \
Item ID Item Name                                                 
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
82      Nirvana                                           $4.90   
145     Fiery Glass Crusader                              $4.58   
92      Final Critic                                      $4.88   
103     Singed Scalpel                                    $4.35   

                                                      Total Purchase Value  
Item ID Item Name                                                           
178     Oathbreaker, Last Hope of the Breaking Storm                 50.76  
82      Nirvana                                                      44.10  
145     Fiery Glass Crusader                                         41.22  
92      Final Critic                                                 39.04  
103     Singed Scalpel                                               34.80

In [219]:
# Check the data distribution of item price
purchase_data['Price'].describe()

count    780.000000
mean       3.050987
std        1.169549
min        1.000000
25%        1.980000
50%        3.150000
75%        4.080000
max        4.990000
Name: Price, dtype: float64

In [220]:
# Reorganize the columns
Profitable_Items_df = Profitable_Items_df[['Purchase Count','Item Price','Total Purchase Value']]
# Format the value
Profitable_Items_df['Total Purchase Value'] = Profitable_Items_df['Total Purchase Value'].map("${:.2f}".format)

In [221]:
Profitable_Items_df.head()

Purchase Count  \
Item ID Item Name                                                      
178     Oathbreaker, Last Hope of the Breaking Storm              12   
82      Nirvana                                                    9   
145     Fiery Glass Crusader                                       9   
92      Final Critic                                               8   
103     Singed Scalpel                                             8   

                                                     Item Price  \
Item ID Item Name                                                 
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
82      Nirvana                                           $4.90   
145     Fiery Glass Crusader                              $4.58   
92      Final Critic                                      $4.88   
103     Singed Scalpel                                    $4.35   

                                                     Total Purchase Value  
Item ID Item Name                                                          
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
82      Nirvana                                                    $44.10  
145     Fiery Glass Crusader                                       $41.22  
92      Final Critic                                               $39.04  
103     Singed Scalpel                                             $34.80

## Conclusion:
      This was a very interesting data analysis. We came out with some very interesting facts about the gaming purchase. After this analysis we can conclude following:
      
      1. The 3 most profitable items are "Oathbreaker, Last Hope of the Breaking Storm","Fiery Glass Crusader', "Nirvana' which are also the most popular items for players. What's more, "Oathbreaker, Last Hope of the Breaking Storm" is both the most popular and most profitable item.
  
      2. Of all the 5 most profitable items, their prices fall between 75%-100%($4.08-$4.09), which indicate the more expensive items bring more profits for company than cheaper items.
      
      3. Although proportion of male players (84%) is way higher than female players(14%). But female's average spending ($3.20) is higher than male players'($3.01). Besides, the second top spender is a female player (SN is Chamjask73). All the evidences indicate the great potential of the women's gaming market and we cann't ignore it. 	
      

      
    